pennylane/_device.py

```diff
@@ -752,6 +752,13 @@ def batch_transform(self, circuit: QuantumTape):
         # device property present in braket plugin
         use_grouping = getattr(self, "use_grouping", True)
 
+        if any(isinstance(m, MidMeasureMP) for m in circuit.operations):
+            circuit = (
+                qml.defer_measurements(circuit)
+                if not self.capabilities().get("supports_mid_measure", False)
```

Comment:
are we ok with adding more to the device capabilities dict? wondering if there's a better way to register this conditionally (for the old device, of course)

Reply:
I think if that's the plan, it's probably out of the scope of this PR. Moreover, the old devices are going away soon so I don't know if this would be something worth doing.

Reply:
Maybe we could modify Line 65 to be:
```
    _capabilities = {"model": None, "supports_broadcasting": False, "supports_mid_measure": False}
```